### ISSUE TITLE
Cursor jumps around while deleting emoji

### ISSUE THREAD

**Issue by miciasto** [typora/typora-issues#5943]
- [x] Searched existing issues to avoid creating duplicates.
- [x] Confirmed that it can be reproduced in built-in themes without customized css.
- [x] Searched http://support.typora.io/
----
**Describe the bug**
Cursor jumps around while deleting emoji.

**To Reproduce**
1. Enter the following text on a new line: 
```
:smile:
```
2. Move the cursor to the start of the line
3. Press the delete key 3 times

**Observed**: 
```
sile:
```

**Expected**:
```
ile:
```

**Desktop:** OS: Windows 11 64 bit
**Typora Version**: 1.8.3-dev

**Comment by miciasto** [comment]
May be a related issue?

**To Reproduce**
1. Enter the following text on a new line: 
```
==:smile:==
```
2. Using the arrow keys (not the mouse), move the cursor to imcmediately before the first : character
3. Press the delete key 7 times

**Observed**:
```
==:smile:==
```
that is, nothing happened, it is as if the delete key does not work.

**Expected**:
```
====
```

**Comment by miciasto** [comment]
Also may be related.  (let me know if you prefer me to report these in separate issues):

----
**Describe the bug**
Cursor moves to incorrect position after entering square brackets.

**To Reproduce**
1. Enter the following text on a new line (note there is a space character at the end of the line): 
```
:smile: [in] 
```
2. Move the cursor to the end of the line.
3. Enter the text: [out]

**Observed**: 
```
:smile: [in] out[]
```

**Expected**:
```
:smile: [in] [out]
```

**Desktop:** OS: Windows 11 64 bit
**Typora Version**: 1.8.3-dev

**Comment by abnerlee** [comment]
should be fixed in 1.8.6

**Comment by miciasto** [comment]
Tested on 1.8.8 and confirmed issue is 99% resolved.  One press of the delete key is ignored but it works if you hit it again.

Thank you
